src/mono/mono/utils/mono-mmap-wasm.c

```diff
@@ -37,22 +35,7 @@
 int
 mono_pagesize (void)
 {
-	if (mono_opt_wasm_mmap)
-		return MWPM_PAGE_SIZE;
-
-	static int saved_pagesize = 0;
-
-	if (saved_pagesize)
-		return saved_pagesize;
-
-	// Prefer sysconf () as it's signal safe.
-#if defined (HAVE_SYSCONF) && defined (_SC_PAGESIZE)
-	saved_pagesize = sysconf (_SC_PAGESIZE);
-#else
-	saved_pagesize = getpagesize ();
-#endif
-
-	return saved_pagesize;
+	return 16384;
```

Comment:
does the loop that's based on SB_MAX_SIZE handle this pagesize correctly?

Reply:
I removed the ifdef and we will see if those asserts still pass. They should as mono_pagesize is 16K and `LOCK_FREE_ALLOC_SB_MAX_SIZE` is 32*16K

Reply:
So that didn't work and I reverted that and keep the ifdef.

Reply:
OK. It might be nice to add a comment near the broken loop/asserts that explains the current state of the code, in case someone other than you and me encounters it later